src/jit/codegenarmarch.cpp

```diff
@@ -1968,6 +1964,8 @@ void CodeGen::genJmpMethod(GenTreePtr jmp)
             continue;
         }
 
+        const unsigned varNum = compiler->lvaTable.GetLclNum(varDsc);
+
```

Comment:
The old code needed to do `varDsc = compiler->lvaTable + varNum;`, so in this case it's a wash. I think the `foreach` style is a bit more bulletproof, hence the change.